File: TwoSum/twosum.js
class Solution {
    /**
     * @param {number[]} nums
     * @param {number} target
     * @return {number[]}
     */
    twoSum(nums, target) {
    const map = new Map()
    for(let i=0; i<nums.length; i++){
        const complement = target - nums[i] 
        const index = map.get(complement)
        const findTarget = map.has(complement)
        if(findTarget){
            return [index, i]
        }
        map.set(nums[i], i)
    }
    return [-1,-1]
    }
}

File: TwoSum/twosum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        x=len(nums)
        y={}
        for i in range(len(nums)):
            complement = target - nums[i]
            if complement in y:
                return [y[complement], i]
            y[nums[i]] = i
        